src/Service/ProducerSideServiceImpl.cpp

```diff
@@ -348,7 +348,7 @@ void ProducerSideServiceImpl::ReceiveEventsThread(
 
     switch (request.event_case()) {
       case orbit_grpc_protos::ReceiveCommandsAndSendEventsRequest::kBufferedCaptureEvents: {
-        absl::MutexLock lock{&capture_event_buffer_mutex_};
+        absl::ReaderMutexLock lock{&capture_event_buffer_mutex_};
```

Comment:
This is correct and in line with the use of `absl::WriterMutexLock` when setting `capture_event_buffer_`, but note that `AddEvent` is called below and it's not `const`. That's fine because of the internal synchronization of `CaptureEventProducer`, but this needs a comment, it took me a while to understand why this change was correct.

Something like
"capture_event_buffer_mutex_ only protects capture_event_buffer_ as a pointer and not the pointee. But the CaptureEventProducer itself is internally synchronized, hence why it's fine to call AddEvent below while only holding a ReaderMutexLock."